addons/binding/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/handler/core/TelldusCoreDeviceController.java

```diff
@@ -0,0 +1,348 @@
+                } else if (command instanceof PercentType) {
+                    dim(device, (PercentType) command);
+                } else if (command instanceof IncreaseDecreaseType) {
+                    increaseDecrease(device, ((IncreaseDecreaseType) command));
+                }
+            } else if (device instanceof SwitchableDevice) {
+                if (command == OnOffType.ON) {
+                    if (isdimmer) {
+                        logger.info("Turn off first in case it is allready on");
+                        turnOff(device);
+                        checkLastAndWait(resendInterval);
+                    }
+                    turnOn(device);
+                } else if (command == OnOffType.OFF) {
+                    turnOff(device);
+                }
+            } else {
+                logger.warn("Cannot send to " + device);
+            }
+        }
+    }
+
+    private void increaseDecrease(Device dev, IncreaseDecreaseType increaseDecreaseType) throws TellstickException {
+        String strValue = ((TellstickDevice) dev).getData();
+        double value = 0;
+        if (strValue != null) {
+            value = Double.valueOf(strValue);
+        }
+        int percent = (int) Math.round((value / 255) * 100);
+        if (IncreaseDecreaseType.INCREASE == increaseDecreaseType) {
+            percent = Math.min(percent + 10, 100);
+        } else if (IncreaseDecreaseType.DECREASE == increaseDecreaseType) {
+            percent = Math.max(percent - 10, 0);
+        }
+
+        dim(dev, new PercentType(percent));
+    }
+
+    private void dim(Device dev, PercentType command) throws TellstickException {
+        double value = command.doubleValue();
+
+        // 0 means OFF and 100 means ON
+        if (value == 0 && dev instanceof SwitchableDevice) {
+            ((SwitchableDevice) dev).off();
+        } else if (value == 100 && dev instanceof SwitchableDevice) {
+            ((SwitchableDevice) dev).on();
+        } else if (dev instanceof DimmableDevice) {
+            long tdVal = Math.round((value / 100) * 255);
+            ((DimmableDevice) dev).dim((int) tdVal);
+        } else {
+            throw new RuntimeException("Cannot send DIM to " + dev);
+        }
+    }
+
+    private void turnOff(Device dev) throws TellstickException {
+        if (dev instanceof SwitchableDevice) {
+            ((SwitchableDevice) dev).off();
+        } else {
+            throw new RuntimeException("Cannot send OFF to " + dev);
+        }
+    }
+
+    private void turnOn(Device dev) throws TellstickException {
+        if (dev instanceof SwitchableDevice) {
+            ((SwitchableDevice) dev).on();
+        } else {
+            throw new RuntimeException("Cannot send ON to " + dev);
+        }
+    }
+
+    private void checkLastAndWait(long resendInterval) {
+        while ((System.currentTimeMillis() - lastSend) < resendInterval) {
+            logger.info("Wait for " + resendInterval + " millisec");

```

Comment:
reduce to debug and change to parameterized logging instead of string concatenation